src/metadataTextTracks.js

```diff
@@ -0,0 +1,128 @@
+/**
+* This feature allows metadata text tracks to be manipulated once available, @see process.
+* It also allows ad implementations to leverage ad cues coming through
+* text tracks, @see processAdTrack
+**/
+
+import videojs from 'video.js';
+
+const metadataTextTracks = {};
+
+/**
+* This feature allows metadata text tracks to be manipulated once they are available,
+* usually after the 'loadstart' event is observed on the player
+* @param player A reference to a player
+* @param processTrack A method that performs some operations on a metadata text track
+**/
+metadataTextTracks.process = function(player, processTrack) {
+  const tracks = player.textTracks();
+  const prepareTrack = function(track) {
+    if (track.kind === 'metadata') {
+      // Make sure track is 'hidden' rather than 'disabled' as it may have already
+      // existed and had been 'disabled' during a playlist change
+      track.mode = 'hidden';
+      processTrack(player, track);
+    }
+  };
+
+  // Text tracks are available
+  if (tracks.length > 0) {
+    for (let i = 0; i < tracks.length; i++) {
+      const track = tracks[i];
+      prepareTrack(track);
```

Comment:
This calls `prepareTrack` for each text track on the player and it in turn enables each one. However, we don't necessarily want to enable all the metadata text tracks. There may be old ones around we don't want enabled.